src/azure/cli/commands/arm.py

```diff
@@ -226,6 +252,11 @@ def __call__(self, parser, namespace, values, option_string=None):
     cmd = CliCommand(name, handler, simple_output_query=simple_output_query)
     cmd.arguments.update(set_arguments)
     cmd.arguments.update(get_arguments)
+    if function_arguments:
+        cmd.arguments.update(function_arguments)
+    cmd.arguments.pop('instance', None) # inherited from custom_function(instance, ...)
+    cmd.arguments.pop('parent', None)
+    cmd.arguments.pop('expand', None) # possibly inherited from the getter
```

Comment:
They accomplish the same thing, though preventing the argument from being added to the argument space to begin with is a better approach than adding it and trying to suppress it. Isn't this why you did the same thing for the setter's parameters arg?

---
In reply to: [76099240](https://github.com/Azure/azure-cli/pull/735#discussion_r76099240) [](ancestors = 76099240,76079251,75964122)